src/attribute.cpp

```diff
@@ -0,0 +1,72 @@
+#include "attribute.h"
+#include "attribute_types.h"
+#include <iostream>
+
+Attribute::Attribute(uint8_t type)
+    : m_type(type)
+{
+}
+
+String::String(uint8_t type)
+        : Attribute(type)
+{
+}
+
+std::string String::value() const
+{
+    return "1"; //temporarily
+}
+
+std::string typeToString(int type)
+{
+    switch (type)
+    {
+        case 1: return "USER_NAME";
```

Comment:
If you have `attribute_types.h` with names assigned to all attribute types, why do you still use numeric values?